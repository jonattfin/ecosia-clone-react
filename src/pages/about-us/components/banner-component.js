import { Button } from '@mui/material';

export default function Component({ language, styles }) {
  return (
    <section className={styles['everyone-power-section']}>
      <div className={styles.title}>We believe in everyone's power to do good</div>
      <div>
        <Button variant="contained">Add Ecosia to Firefox</Button>
      </div>
      <div></div>
    </section>
  )
}
