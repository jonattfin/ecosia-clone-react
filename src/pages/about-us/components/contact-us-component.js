import { TextField, Button, MenuItem } from '@mui/material';

import styles from './styles.module.scss';

export default function Component() {
  return (
    <section className={styles['contact-us-section']}>
      <div className={styles.title}>Contact us</div>
      <div className={styles.subtitle}>For questions about Ecosia check our FAQ first. <br />If you don't see what you are looking for, drop us a line!</div>
      <div className={styles.form}>
        <TextField variant="outlined" label="Message" multiline rows={5}></TextField>
        <TextField variant="outlined" label="E-mail" />
        <TextField variant="outlined" label="Subject" select>
          {getFilterOptions().map((option) => (
            <MenuItem key={option} value={option}>
              {option}
            </MenuItem>
          ))}
        </TextField>
        <Button variant="contained">Send</Button>
      </div>
    </section>
  )
}

function getFilterOptions() {
  return ["Subject", "Technical problem", "General inquiry", "Press inquiry"];
}
