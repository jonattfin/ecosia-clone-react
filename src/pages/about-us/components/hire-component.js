
import { Button } from '@mui/material';

import styles from './styles.module.scss';

export default function Component() {
  return (
    <section className={styles['hiring-section']}>
      <div className={styles.title}>We're hiring!</div>
      <div className={styles.subtitle}>Want to help change the world? Let's team up. Apply to help us build great products and services that can make the world a more sustainable place. We can't wait to hear from you.</div>
      <div>
        <Button variant="contained">See openings</Button>
      </div>
    </section>
  )
}
