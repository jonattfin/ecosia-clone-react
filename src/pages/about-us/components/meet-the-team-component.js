
import { useState } from "react";
import { Button } from '@mui/material';

import styles from './styles.module.scss';

export default function Component() {
  const [play, setPlay] = useState(false);

  return (
    <section className={styles['meet-the-team-section']}>
      {!play && (
        <>
          <div className={styles.title}>Step inside!</div>
          <div className={styles.subtitle}>Meet the humans who make Ecosia and learn what it's like to work here.</div>
          <div>
            <Button variant="contained" onClick={() => setPlay(true)} >Play</Button>
          </div>
        </>
      )}
      {play && (
        <div className={styles['movie']}>
          <video className={styles['movie__wrapper']} controls autoPlay={true}>
            <source src="https://d3fme2ivr1xlgj.cloudfront.net/240719_Ecosia_hiringvideo_subbed_titled.mp4" type="video/mp4" />
          </video>
        </div>
      )}
    </section>
  )
}
