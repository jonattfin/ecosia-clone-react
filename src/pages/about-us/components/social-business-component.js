
export default function Component({ language, styles }) {
  return (
    <section className={styles['social-business-section']}>
      <div className={styles.title}>Ecosia is a social business founded in 2009 after a trip around the world</div>
      <div className={styles.column}>
        {getTimePeriods().map((period, index) => (
          <div key={`period_${index}`} className={styles.column__item}>
            <div className={styles.time}>{period.time}</div>
            <div className={styles.subtitle}>{period.title}</div>
            <hr className={styles.line} />
            <div>{period.content}</div>
          </div>
        ))}
      </div>
    </section>
  )
}

function getTimePeriods() {
  return [
    {
      time: 'December 2009',
      title: 'Ecosia is born',
      content: 'Christian founded Ecosia.org after a trip around the world helped him understand the problems of deforestation.'
    },
    {
      time: '2009 - 2011',
      title: 'People\'s choice',
      content: 'Ecosia won several awards for its clever concept and speedy growth in Europe and beyond.'
    },
    {
      time: 'April 2014',
      title: 'First German B Corp',
      content: 'Ecosia was the first German company to become a B Corporation thanks to its social business model.'
    },
    {
      time: 'April 2018',
      title: '25 million trees',
      content: 'A planting milestone! That same year, Ecosia also builds its own solar energy plant to power every search.'
    },
  ]
}
