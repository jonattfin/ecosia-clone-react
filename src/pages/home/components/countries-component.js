
import * as Images from './images';

export default function Component({ language, styles }) {
  return (
    <section className={styles['countries-section']}>
      <div className={styles['above-title']}>
        {language['where-do-you-plant-trees']}
      </div>
      <div className={styles.title}>
        {language['we-plant-30-countries']}
      </div>
      <div className={styles['countries-grid']}>
        <div className={styles['grid-item']}>
          <div><img className={styles.country} src={Images.BrazilImage} alt="brazil" /></div>
          <div className={styles['grid-item__title']}>Brazil</div>
          <div className={styles['grid-item__subtitle']}>
            {language['we-plant-in-brazil']}
          </div>
        </div>
        <div className={styles['grid-item']}>
          <img className={styles.country} src={Images.BurkinaFasoImage} alt="burkina faso" />
          <div className={styles['grid-item__title']}>Burkina Faso</div>
          <div className={styles['grid-item__subtitle']}>{language['we-plant-in-burkina-faso']}</div>
        </div>
        <div className={styles['grid-item']}>
          <img className={styles.country} src={Images.IndonesiaImage} alt="indonesia" />
          <div className={styles['grid-item__title']}>Indonesia</div>
          <div className={styles['grid-item__subtitle']}>{language['we-plant-in-indonesia']}</div>
        </div>
      </div>
    </section >
  )
}
