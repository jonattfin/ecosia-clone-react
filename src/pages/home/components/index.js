import PieComponent from './responsive-pie';

const Components = {
  PieComponent
}

export default Components;
