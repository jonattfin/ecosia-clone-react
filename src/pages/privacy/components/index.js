export { default as ProtectPrivacyComponent } from './protect-privacy-component';
export { default as StatementsComponent } from './statements-component';
