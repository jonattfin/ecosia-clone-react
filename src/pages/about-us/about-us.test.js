import React from 'react';
import renderer from 'react-test-renderer';

import * as Components from './components';
import styles from './styles.module.scss';

it('<Components.BannerComponent/> renders correctly', () => {
  const tree = renderer
    .create(<Components.BannerComponent {...{ styles }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// it('<Components.ContactUsComponent/> renders correctly', () => {
//   const tree = renderer
//     .create(<Components.ContactUsComponent {...{ styles }} />)
//     .toJSON();
//   expect(tree).toMatchSnapshot();
// });

it('<Components.HireComponent/> renders correctly', () => {
  const tree = renderer
    .create(<Components.HireComponent {...{ styles }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('<Components.MeetTheTeamComponent/> renders correctly', () => {
  const tree = renderer
    .create(<Components.MeetTheTeamComponent {...{ styles }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('<Components.SocialBusinessComponent/> renders correctly', () => {
  const tree = renderer
    .create(<Components.SocialBusinessComponent {...{ styles }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


