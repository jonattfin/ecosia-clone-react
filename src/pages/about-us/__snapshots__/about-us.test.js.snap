// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Components.BannerComponent/> renders correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-xl-12 css-1idn90j-MuiGrid-root"
  >
    <div
      className="css-1637b1s"
    >
      <div
        className="css-1vzsge8"
      >
        We believe in everyone's power to do good
      </div>
      <div>
         
      </div>
      <div>
        <button
          className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Add Ecosia to Firefox
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Components.HireComponent/> renders correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-xl-12 css-1idn90j-MuiGrid-root"
  >
    <div
      className="css-1hsx150"
    >
      <div
        className="css-1vzsge8"
      >
        We're hiring!
      </div>
      <div
        className="css-18ogej"
      >
        Want to help change the world? Let's team up. 
        <br />
        Apply to help us build great products and services that can make the world a more sustainable place. 
        <br />
        We can't wait to hear from you.
      </div>
      <div>
        <button
          className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          See openings
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Components.MeetTheTeamComponent/> renders correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-xl-12 css-1idn90j-MuiGrid-root"
  >
    <div
      className="css-14wcu1j"
    >
      <div
        className="css-1vzsge8"
      >
        Step inside!
      </div>
      <div
        className="css-18ogej"
      >
        Meet the humans who make Ecosia and learn what it's like to work here.
      </div>
      <div>
        <button
          className="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-ke5b6m-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Play
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Components.SocialBusinessComponent/> renders correctly 1`] = `
<div
  className="css-flwcnq"
>
  <div
    className="css-1vzsge8"
  >
    Ecosia is a social business founded in 2009 after a trip around the world
  </div>
  <div
    className="css-ef7oun"
  >
    <div>
      <div
        className="css-1wjqaby"
      >
        December 2009
      </div>
      <div
        className="css-1yltisf"
      >
        Ecosia is born
      </div>
      <div
        className="css-13vo6hn"
      />
      <div
        className="css-1yltisf"
      >
        Christian founded Ecosia.org after a trip around the world helped him understand the problems of deforestation.
      </div>
    </div>
    <div>
      <div
        className="css-1wjqaby"
      >
        2009 - 2011
      </div>
      <div
        className="css-1yltisf"
      >
        People's choice
      </div>
      <div
        className="css-13vo6hn"
      />
      <div
        className="css-1yltisf"
      >
        Ecosia won several awards for its clever concept and speedy growth in Europe and beyond.
      </div>
    </div>
    <div>
      <div
        className="css-1wjqaby"
      >
        April 2014
      </div>
      <div
        className="css-1yltisf"
      >
        First German B Corp
      </div>
      <div
        className="css-13vo6hn"
      />
      <div
        className="css-1yltisf"
      >
        Ecosia was the first German company to become a B Corporation thanks to its social business model.
      </div>
    </div>
    <div>
      <div
        className="css-1wjqaby"
      >
        April 2018
      </div>
      <div
        className="css-1yltisf"
      >
        25 million trees
      </div>
      <div
        className="css-13vo6hn"
      />
      <div
        className="css-1yltisf"
      >
        A planting milestone! That same year, Ecosia also builds its own solar energy plant to power every search.
      </div>
    </div>
  </div>
  <div>
     
  </div>
</div>
`;
